[iOS] Offline help as mirror of online help.keyman.com iOS help

jahorton opened the pull request:
Based on #2088.

In an effort to reduce the WETness (*w*rite *e*verything *t*wice) of our existing iOS documentation, this PR seeks to near-fully eliminate the maintenance overhead of our offline iOS help.  This is accomplished via a maintenance script (`build-help.sh`) that produces a fully-functional offline mirror of the iOS section of help.keyman.com and its required resources.

We'll settle for manually running it for now, as the script currently requires the live site and CI could run into problems against that.  A simple run-and-confirm is pretty maintenance-lite, so this is fine.

Advice for reviews:  ignore commit https://github.com/keymanapp/keyman/pull/2102/commits/7e206ff41a03a7826176e4ea20560c546d9e1dd9 - it's the result of running the maintenance script.  The other commits contain the real goodies.

For those concerned, the file size of the resulting `OfflineHelp.bundle` file is just shy of 1 MB.  Use of the "bundle" structure means that Xcode doesn't need manual tracking of files within it, unlike the required file tracking outside of a bundle.  (Bundles are tracked as a unit; compare against the changes in ios/keyman/Keyman/Keyman.xcodeproj/project.pbxproj.)  This is a major win for ease of maintenance.

---

If you want to test the offline version without taking your whole machine offline:

- Go to InfoViewController.swift (within the InfoViewController group)
- Search for `reloadKeymanHelp`
- Comment out the `switch` block aside from `loadFromLocal()`

darcywong00 commented on ios/build-help.sh:
```diff
@@ -0,0 +1,36 @@
+HELP_ROOT=keyman/Keyman/resources/OfflineHelp.bundle/Contents/Resources
+
+if [ -z $VERSION ]; then
+  VERSION="12.0"  # Set manually here because alpha versions usually don't have 
```
I think we want to avoid manually updating version strings in build scripts.

Can you normally use  `cat ../resources/VERSION.md`?
Then, if you have access to the environment `$TIER` and $TIER? is "alpha", decrement the version.

jahorton replied:
I thought about it... decrementing a string's not the simplest thing, though.  I'll look into it.

jahorton replied:
OK, got something working.

darcywong00 replied:
Nice! I just remembered you also added a redirect on help.keyman.com for alpha versions to go to the stable version. So you've got it covered on both ends

jahorton commented:
Tried to fix a commit name, since the tweaks weren't actually that 'minor', but that kinda broke the git history, requiring the force-push.  :(

Tweaks:
- Fixed an error message
- Added $TIER=alpha version decrementing.

darcywong00 commented on ios/build-help.sh:
```diff
@@ -0,0 +1,47 @@
+  # More readable:  (major).minor
+  [[ "$VERSION" =~ ([0-9]+)\.[0-9]+ ]]
+  # Construct a decremented version string with .minor set to .0
+  $VERSION="$((${BASH_REMATCH[1]}-1)).0"
```
```suggestion
  VERSION="$((${BASH_REMATCH[1]}-1)).0"
```
